dotnet/src/InternalUtilities/planning/PlanInstrumentation.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.SemanticKernel.Planning;
+
+/// <summary>Surrounds the invocation of a planner with logging and metrics.</summary>
+internal static partial class PlanInstrumentation
+{
+    /// <summary><see cref="Histogram{T}"/> to record plan execution duration.</summary>
+    private static readonly Histogram<double> s_planExecutionDuration =
+        PlanningInstrumentation.Meter.CreateHistogram<double>(
+            name: "sk.planning.run_plan.duration",
+            unit: "s",
+            description: "Duration time of plan execution.");
+
+    /// <summary><see cref="Counter{T}"/> to record planner execution success counts.</summary>
+    private static readonly Counter<int> s_executionSuccess =
+        PlanningInstrumentation.Meter.CreateCounter<int>(
+            name: "sk.planning.run_plan.success",
+            unit: "{invocation}",
+            description: "Measures the number of successful plan executions");
+
+    /// <summary><see cref="Counter{T}"/> to record planner execution failure counts.</summary>
+    private static readonly Counter<int> s_executionFailure =
+        PlanningInstrumentation.Meter.CreateCounter<int>(
+            name: "sk.planning.run_plan.failure",
+            unit: "{invocation}",
+            description: "Measures the number of failed plan executions");
```

Comment:
Do we need these? This information should all be available via the s_planExecutionDuration histogram: we can just add an error.type tag when the counter is failure.  This is what we do for function invocation.

Reply:
You're right. We can filter the measurements in Azure monitor. Removed.